ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/SyncTargetManager.java

```diff
@@ -120,18 +127,48 @@ protected CompletableFuture<Optional<EthPeer>> selectBestAvailableSyncTarget() {
 
   private CompletableFuture<Optional<EthPeer>> confirmPivotBlockHeader(final EthPeer bestPeer) {
     final BlockHeader pivotBlockHeader = fastSyncState.getPivotBlockHeader().get();
-    final RetryingGetHeaderFromPeerByNumberTask task =
-        RetryingGetHeaderFromPeerByNumberTask.forSingleNumber(
-            protocolSchedule,
-            ethContext,
-            metricsSystem,
-            pivotBlockHeader.getNumber(),
-            MAX_QUERY_RETRIES_PER_PEER);
-    task.assignPeer(bestPeer);
-    return ethContext
-        .getScheduler()
-        // Task is a retrying task. Make sure that the timeout is long enough to allow for retries.
-        .timeout(task, Duration.ofSeconds(MAX_QUERY_RETRIES_PER_PEER * SECONDS_PER_REQUEST + 2))
+    CompletableFuture<List<BlockHeader>> headersFuture;
+    if (config.isPeerTaskSystemEnabled()) {
+      headersFuture =
+          ethContext
+              .getScheduler()
+              .scheduleServiceTask(
+                  () -> {
+                    GetHeadersFromPeerTask task =
+                        new GetHeadersFromPeerTask(
```

Comment:
This should use the same retries as the original task from `MAX_QUERY_RETRIES_PER_PEER`

Reply:
I think I ignore this originally because that constant is already the same as the default we're using for the peer task system, but it makes for it to be explicitly set to this constant